modules/ocf_backups/files/prune-old-backups

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import json
+import shlex
+import subprocess
+import sys
+import textwrap
+
+RETENTION_TIME = datetime.timedelta(days=180) # 6 months
+
+BACKUP_PREFIX = 'ocf-backup-'
+BOX_FTP = 'ftps://ftp.box.com:990'
+CREDS_PATH = '/opt/share/backups/box-creds.json'
+
+LFTP_INIT = '''
+    set ftps:initial-prot ""
+    set ftp:ssl-force true
+    set ftp:ssl-protect-data true
+    open {box_ftp}
+    user {email} {passwd}
+'''
+
+def parse_date(dtstr):
+    """Converts a 'YYYY-MM-DD' formatted string to a datetime.date object."""
+    return datetime.datetime.strptime(dtstr, "%Y-%m-%d").date()
+
+def list_items(creds):
+    """List all items at the root of the Box.com folder."""
+    lftp_cmd = (LFTP_INIT + 'renlist').format(
+        box_ftp=BOX_FTP,
+        email=creds['email'],
+        passwd=creds['password'],
+    )
+
+    p = subprocess.run(
+        ['lftp', '--norc'],
+        input=lftp_cmd.encode(),
+        stdout=subprocess.PIPE,
+    )
+
+    p.check_returncode()
+
+    return p.stdout.decode().splitlines()
+
+def delete_items(creds, items, quiet):
+    lftp_cmd = (LFTP_INIT + 'rm -r {items}').format(
+        box_ftp=BOX_FTP,
+        email=creds['email'],
+        passwd=creds['password'],
+        items=' '.join(shlex.quote(item) for item in items)
+    )
+
+    if quiet:
+        stdout_pipe = subprocess.DEVNULL
+    else:
+        stdout_pipe = None
+
+    p = subprocess.run(
+        ['lftp', '--norc'],
+        input=lftp_cmd.encode(),
+        stdout=stdout_pipe
+    ).check_returncode()
+
+def is_old_backup(filename):
+    """Returns true if this backup is old enough to be deleted."""
+    start, _, end = filename.partition(BACKUP_PREFIX)
+
+    # Not a backup
+    if start:
+        return False
+
+    try:
+        backup_date = parse_date(end)
+    except ValueError:
+        return False
+
+    return datetime.date.today() - backup_date > RETENTION_TIME
+
+def main(args):
+    with open(CREDS_PATH) as f:
+        creds = json.load(f)
+
+    items = set(list_items(creds))
+
+    to_delete = {item for item in items if is_old_backup(item)}
+
+    if not args.quiet:
+        print('Items to keep:')
+        print(textwrap.indent('\n'.join(sorted(items - to_delete)), '  '))
+        print()
+
+        print('Items to delete:')
+        print(textwrap.indent('\n'.join(sorted(to_delete)), '  '))
+
+    if not args.dry_run:
+        delete_items(to_delete)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Prune old Box.com backups')
+    parser.add_argument(
+        '-n', '--dry-run',
+        action='store_true',
+        help='Show which backups would be deleted, but do not delete.'
+    )
+    parser.add_argument(
+        '-q', '--quiet',
+        action='store_true',
+        help='do not output progress or information to stdout',
+    )
+
+    args = parser.parse_args()
+    sys.exit(main(args))
```

Comment:
`main` doesn't return anything, so I don't see why we are calling `sys.exit` on the result, which will always be `None`...

Reply:
This is standard for pretty much all of our scripts.

Reply:
At least in the scripts I looked at, `main` returns some exit code (which is actually not great, I much prefer explicit `sys.exit`ing).

In other scripts where it does not, this is not used (https://github.com/ocf/puppet/blob/master/modules/ocf_desktop/files/xsession/auto-lock, https://github.com/ocf/puppet/blob/master/modules/ocf_desktop/files/xsession/lab-close-notify). Its pointless to call sys.exit here, as the script will exit with 0 anyway (or it won't, which this won't change).

Reply:
Alright fair enough